lib/src/main/scala/spinal/lib/bus/tilelink/WidthAdapter.scala

```diff
@@ -64,25 +66,31 @@ class WidthAdapter(ip : BusParameter,
       buffer.first := wordLast
       when(buffer.first) {
         buffer.args.assignSomeByName(src.payload)
-        buffer.corrupt := False
+        if(src.withData) buffer.corrupt := False
         if(src.withDenied) buffer.denied := False
         if(src.withMask) buffer.mask.foreach(_ := 0)
         if(src.withSink) buffer.sink := src.sinkNull
       }
 
-      if(src.withMask) {
-        buffer.data(sel) := src.data
-        buffer.mask(sel) := src.maskNull
-      } else {
-        val maskRange = log2Up(src.p.dataBytes)+1 to log2Up(dst.p.dataBytes)
-        val mask = maskRange.map(src.size >= _).asBits().asUInt
-        for(i <- 0 until ratio){
-          when(((sel ^ i) & mask) === 0){
-            buffer.data(i) := src.data
+      if(src.withData) {
+        if(src.withMask) {
+          buffer.data(sel) := src.data
+          buffer.mask(sel) := src.maskNull
+        } else {
+          if(src.size.maxValue > log2Up(src.p.dataBytes)) {
```

Comment:
Hi ^^

I'm not sure to understand this line `if(src.size.maxValue > log2Up(src.p.dataBytes)) {`

In which case you need it ?

Reply:
Below are two examples that seem like corner cases.

``` scala
object WidthAdapterExample extends App{
  SpinalVerilog{
    new Component {
      def bp(dataWidth: Int, sizeBytes: Int) = {
        NodeParameters(
          M2sParameters(
            addressWidth = 12,
            dataWidth = dataWidth,
            masters = List(
              M2sAgent(
                name = null,
                mapping = List(
                  M2sSource(
                    id = SizeMapping(0, 16),
                    emits = M2sTransfers(
                      get = SizeRange(sizeBytes),
                      putPartial = SizeRange(sizeBytes),
                    )
                  )
                )
              )
            )
          )
        ).toBusParameter()
      }

      val ip = bp(dataWidth = 32, sizeBytes = 2)
      val op = bp(dataWidth = 16, sizeBytes = 2)

      // val ip = bp(dataWidth = 128, sizeBytes = 8)
      // val op = bp(dataWidth = 64, sizeBytes = 8)

      new tilelink.WidthAdapter(ip, op)
    }
  }
}
```

For the first example
``` scala
      val ip = bp(dataWidth = 32, sizeBytes = 2)
      val op = bp(dataWidth = 16, sizeBytes = 2)
```

The type of `src.size` is `UInt(1 bits)`, while `log2Up(src.p.dataWidth)` is `1`.